### DISCUSSION TITLE
Best practice question for asserting dynamic date in query

### DISCUSSION THREAD

**Discussion by chris-cotech** [prisma/prisma#24563] [Q&A] [answered]
### Question

Hi Prisma Community,

I know there is a new feature to use client.{tableName}.fields.field instead of rawQuery. 

I would like to know if something like this is possible:

```js
const currentDate = new Date()

await prismaGenbai.sprint.findMany({
      where: {
        teamId: id,
        sprintPhase: { NOT: { phaseName: 'Complete' } },
        startDate: {
          lt: currentDate.setDate(
            currentDate.getDate() -
              Number(prismaGenbai.sprint.fields.durationInDays)
          )
        }
      },
      select: {
        id: true,
        name: true
      }
    })
```

Currently the "lt" (less than) object is displaying a typescript error with the following:

Type 'number' is not assignable to type 'string | Date | FieldRef<"Sprint", "DateTime"> | undefined'.ts(2322)
(property) lt?: string | Date | Prisma.FieldRef<"Sprint", "DateTime"> | undefined

![image](https://github.com/prisma/prisma/assets/90394436/6d3b06f2-bfe7-4b22-98bc-5f570236c5fe)

I believe that it makes sense that this wouldn't work because new Date() is defined prior to receiving the durationInDays field from the database (I could be wrong). Does Prisma ORM currently support exposing database functions like `SELECT DATEADD(day, 5, GETDATE());` so that I would be able to replace the 5 with a field name for example?

Appreciate your insights!

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @chris-cotech 👋 

Thank you for raising this question.

Prisma ORM does not currently support directly exposing database functions like `SELECT DATEADD(day, 5, GETDATE());` in the Prisma Client query syntax. There is an open GitHub issue discussing the support for default database functions in the Prisma schema. You can follow and upvote the issue [here](https://github.com/prisma/prisma/issues/7815). You might need to handle the date calculation logic in your application code before passing it to the Prisma Client query.

**Comment by chris-cotech** [comment]
Hi @ludralph 

Thanks for the answer, I upvoted the feature 👌 . Prisma is shaping up to be a nice ORM. Nice one.
